packages/react-components/src/molecules/__tests__/Modal.test.tsx

```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import Modal from '../Modal';
+
+it('renders children', () => {
+  const { container } = render(<Modal>test123</Modal>);
+  expect(container.textContent).toEqual('test123');
+});
```

Comment:
If you want to try out writing a browser I'd have an idea here (this is totally optional, only if you wanna try it out):
test that the overlay has the same bounding rect as the viewport